Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -943,6 +943,79 @@ public AndConstraint<TAssertions> ContainInOrder(params T[] expected)
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Expects the current collection to contain the specified elements in the exact same order, and to be consecutive.
+    /// using their <see cref="object.Equals(object)" /> implementation.
+    /// </summary>
+    /// <param name="expected">An <see cref="IEnumerable{T}"/> with the expected elements.</param>
+    public AndConstraint<TAssertions> ContainInConsecutiveOrder(params T[] expected)
+    {
+        return ContainInConsecutiveOrder(expected, string.Empty);
+    }
+
+    /// <summary>
+    /// Expects the current collection to contain the specified elements in the exact same order, and to be consecutive.
+    /// </summary>
+    /// <remarks>
+    /// Elements are compared using their <see cref="object.Equals(object)" /> implementation.
+    /// </remarks>
+    /// <param name="expected">An <see cref="IEnumerable{T}"/> with the expected elements.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="expected"/> is <c>null</c>.</exception>
+    public AndConstraint<TAssertions> ContainInConsecutiveOrder(IEnumerable<T> expected, string because = "",
+        params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(expected, nameof(expected), "Cannot verify ordered containment against a <null> collection.");
+
+        bool success = Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject is not null)
+            .FailWith("Expected {context:collection} to contain {0} in order{reason}, but found <null>.", expected);
+
+        if (success)
+        {
+            IList<T> expectedItems = expected.ConvertOrCastToList();
+            IList<T> actualItems = Subject.ConvertOrCastToList();
+
+            int subjectIndex = 0;
+
+            Func<T, T, bool> areSameOrEqual = ObjectExtensions.GetComparer<T>();
+            int index;
```

Comment:
♻️ You can move the declaration in the `for` loop. It'll helps to make the method less long (and noisy).

Reply:
I'm not sure what you mean here, the declarations here need to be outside of the for loop to work

Reply:
Why? It's only needed within the block scope created by the `for` loop.

Reply:
Compiler error AV1530 - if you declare the index variable in the loop you cannot update it in the loop itself. Which we need to do to reset it should we have to start searching from the start of the unexpected items again.

(Of course, that does raise the question of if the loop is the right way to solve the problem 🤔 - but I think that it is the cleanest)

The 'subjectIndex` and `highestIndex` need to be outside of the loop as they shouldn't be recreated/reset each loop.

Reply:
Ah, that wasn't obvious at all. Updating a loop variable is quite ugly indeed. But I see now how it works.

Reply:
Would you rather that this is really obvious?

We could declare the index in the loop and disable VS1530 for the loop itself... I see that is done elsewhere for other reasons.

Reply:
I think it would make it clearer that you're doing that.

Reply:
Having that code to disable the rule annoyed me so I spent some time thinking about it and figured out how to rewrite it to remove the index manipulation. I'm on the fence as to which is easier to read, let me know which you prefer (I'll revert if you want 😅)